docs/csharp/whats-new/tutorials/primary-constructors.md

```diff
@@ -0,0 +1,110 @@
+---
+title: Declare and use primary constructors in classes and structs
+description: "Learn how and when to declare primary constructors in your class and struct types. Primary constructors provide concise syntax to declare constructor parameters available anywhere in your type."
+ms.date: 05/26/2023
+---
+# Tutorial: Explore primary constructors
+
+C# 12 introduces [*primary constructors*](../../programming-guide/classes-and-structs/instance-constructors.md#primary-constructors), a concise syntax to declare constructors whose parameters are available anywhere in the body of the type.
+
+In this tutorial, you learn how to:
+
+> [!div class="checklist"]
+>
+> - When to declare a primary constructor on your type
+> - How to call call primary constructors from other constructors
+> - How to use primary constructor parameters in members of the type
+> - Where primary constructor parameters are stored
+
+## Prerequisites
+
+You need to set up your machine to run .NET 8 or later, including the C# 12 or later compiler. The C# 12 compiler is available starting with [Visual Studio 2022 version 17.7](https://visualstudio.microsoft.com/vs) or the [.NET 8 SDK](https://dotnet.microsoft.com/download).
+
+Key learning experiences for this tutorial:
+
+1. declare a primary constructor when parameters are needed in one or more members.
+    1. initialize properties (same or different type.)
+    1. as a service or dependencies.
+    1. pass to base class with primary constructor
+1. consider when a reasonable can be used in a default ctor.
+1. Use primary ctor parameter as you would use a field.
+
+## Primary constructors
+
+You can add parameters to a `struct` or `class` declaration to create a *primary constructor*. You declare a primary constructor when a struct or class **requires** parameters to initialize an instance. Primary constructor parameters are in scope throughout the class definition. It's important to view primary constructor parameters as *parameters* even though they are in scope throughout the class definition. Several rules clarify that they're parameters:
+
+1. Primary constructor parameters aren't stored directly as fields.
+1. Primary constructors aren't members of the class. For example, a primary constructor parameter named `param` can't be accessed as `this.param`.
+1. Primary constructor parameters are mutable.
+1. Primary constructor parameters don't become properties, except in [`record`](../../language-reference/builtin-types/record.md) types.
+
+These rules are consistent with parameters to any method, including other constructor declarations.
+
+The two most common uses for a primary constructor parameter are:
+
+1. As an argument to a `base()` constructor invocation.
+1. To initialize a member field or property.
```

Comment:
i'd say there are three common usages.  the two above, and also just referencing the parameter directly from within members of hte type.